feat: LEAP-1119: Feature/gridview thumbnail

juliosgarbi opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

### Describe the reason for change
When using multilayered images, the original behaviour was to create a src that separate each image by a ','.

This way, it's not displayable.
![image](https://github.com/HumanSignal/label-studio/assets/26071804/cf8fd01b-fc47-4ada-a97d-fc8ff405a647)

#### What does this fix?
It fixes an issue in the gridview when there is multiple layers image.
![image](https://github.com/HumanSignal/label-studio/assets/26071804/ee02ba17-f716-4e8a-870c-5721d4d9256d)

#### What is the new behavior?
We use the first image of the multilayer images.

#### What is the current behavior?
The current behavior is that in this case, images url are separated by a ',' in the image src.
What display nothing at all


### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [x] unit

### Which logical domain(s) does this change affect?
GridView => Data Exploration

FrsECM committed:
Add Thumbnail + Fix bug with Multilayers

FrsECM committed:
Add etc to gitignore

juliosgarbi committed:
merge with develop

juliosgarbi committed:
fix the file extension

juliosgarbi committed:
fix biome

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9373822923

juliosgarbi committed:
fix biome check

juliosgarbi committed:
fix unit tests for datamanager and add tests for the change

juliosgarbi committed:
Merge branch 'feature/gridview-thumbnail' into fb-leap-1119

juliosgarbi committed:
remove the validation for thumbnail in the grid view, since the naming is not consistent and it can create confusion

juliosgarbi committed:
Merge branch 'feature/gridview-thumbnail' into fb-leap-1119

juliosgarbi committed:
fix biomejs

juliosgarbi committed:
fix biome checks

juliosgarbi committed:
fix biome checks

juliosgarbi committed:
Merge branch 'feature/gridview-thumbnail' into fb-leap-1119

juliosgarbi commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9401367424)
> Successfully merged:  10 files changed, 84 insertions(+), 19 deletions(-)

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-1119'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9401367424

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9401396938

bmartel reviewed (approved):
Nicely done, tests and all 🔥

hlomzik commented on web/libs/datamanager/src/components/MainView/GridView/GridView.jsx:
```diff
@@ -23,14 +23,22 @@ const GridHeader = observer(({ row, selected }) => {
+    let value = getProperty(row, valuePath);
 
-    return <GridDataGroup key={`${row.id}-${index}`} type={field.currentType} value={value} field={field} row={row} />;
+    /**The value is an array...
+     * In this case, we take the first element of the array
+     */
```
```suggestion
    // for multilayered images we display the first one
```
Let's describe why instead of what

juliosgarbi committed:
fix unit tests

juliosgarbi committed:
Merge branch 'feature/gridview-thumbnail' into fb-leap-1119

juliosgarbi commented:
It is an internal PR for a community PR: #5769

It won't be merged; it is just for internal purposes.

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9404316820

jombooth commented on web/libs/datamanager/src/components/MainView/GridView/GridView.jsx:
```diff
@@ -23,14 +23,22 @@ const GridHeader = observer(({ row, selected }) => {
+    /**The value is an array...
+     * In this case, we take the first element of the array
+     */
+    if (Array.isArray(value)) {
```
should we also check the length of the array prior to retrieving its first element, to avoid a potential crash?

juliosgarbi replied:
It's not necessary Jo, since we are using getProperty to get value, if the array is empty it's returning undefined and it's not getting into this if

juliosgarbi committed:
Update web/libs/datamanager/src/components/MainView/GridView/GridView.jsx

Co-authored-by: hlomzik <hlomzik@gmail.com>

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9447742065)
> Successfully merged:  34 files changed, 1556 insertions(+), 219 deletions(-)

hlomzik committed:
Merge branch 'develop' into 'fb-leap-1119'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9447742065

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9447762470

juliosgarbi commented:
Closing this PR, since the original is already merged
